src/stanc/stanc.ml

```diff
@@ -223,19 +211,21 @@ let options =
       , Arg.Set print_info_json
       , " If set, print information about the model." ) ]
 
-let print_deprecated_arg_warning =
+(* To be removed in Stan 2.33 *)
+let removed_arg_errors =
   (* is_prefix is used to also cover the --include-paths=... *)
   let arg_is_used arg =
     Array.mem ~equal:(fun x y -> String.is_prefix ~prefix:x y) Sys.argv arg
   in
-  if arg_is_used "--allow_undefined" then
+  if arg_is_used "--allow_undefined" then (
     eprintf
-      "--allow_undefined is deprecated and will be removed in Stan 2.32.0. \
-       Please use --allow-undefined.\n" ;
-  if arg_is_used "--include_paths" then
+      "--allow_undefined was removed in Stan 2.32.0. Please use \
+       --allow-undefined.\n" ;
+    exit 69 (* EX_UNAVAILABLE in sysexits.h*) ) ;
```

Comment:
Do we document `stanc3` exit codes? Do we use other codes from `sysexits.h`?

I don't know what `sysexits.h` is but googling a bit I found https://man.netbsd.org/sysexits.3 and it looks to me that `EX_DATAERR` may be more appropriate than `EX_UNAVAILABLE` here

|code|description|
|-------|----------|
| EX_DATAERR | The input data was incorrect in some way.  This should only be used for user's data and not system files. |
| EX_UNAVAILABLE | A service is unavailable.  This can occur if a support program or file does not exist.  This can also be used as a catchall message when something you wanted to do does not work, but you do not know why. |

Reply:
Currently the only exit codes we use are 0, 1, and 127 (only used if you don't pass a model on the command line). I figured it could be nice to pass a specific exit code for this failure since then wrappers like cmdstanr/py have the option of providing very specific messages. I'm happy for it to be any one of the numeric codes, though.